apps/src/lib/kits/maker/ui/BoardSetupCheck.jsx

```diff
@@ -0,0 +1,335 @@
+/** @file Maker Board setup checker */
+import React, {Component} from 'react';
+import CircuitPlaygroundBoard from '../CircuitPlaygroundBoard';
+import {ensureAppInstalled, findPortWithViableDevice} from '../portScanning';
+import SetupStep, {
+  HIDDEN,
+  WAITING,
+  ATTEMPTING,
+  SUCCEEDED,
+  FAILED,
+  CELEBRATING,
+} from './SetupStep';
+
+const STATUS_IS_CHROME = 'statusIsChrome';
+const STATUS_APP_INSTALLED = 'statusAppInstalled';
+const STATUS_WINDOWS_DRIVERS = 'statusWindowsDrivers';
+const STATUS_BOARD_PLUG = 'statusBoardPlug';
+const STATUS_BOARD_CONNECT = 'statusBoardConnect';
+const STATUS_BOARD_COMPONENTS = 'statusBoardComponents';
+
+const initialState = {
+  isDetecting: false,
+  caughtError: null,
+  [STATUS_IS_CHROME]: WAITING,
+  [STATUS_APP_INSTALLED]: WAITING,
+  [STATUS_WINDOWS_DRIVERS]: WAITING,
+  [STATUS_BOARD_PLUG]: WAITING,
+  [STATUS_BOARD_CONNECT]: WAITING,
+  [STATUS_BOARD_COMPONENTS]: WAITING,
+};
+
+export default class BoardSetupCheck extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {...initialState};
+  }
```

Comment:
we have transform-class-properties syntax enabled, so you can ditch the constructor function and just have:

```jsx
state = {...initialState}
```